File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.cs
public class Solution
{
     
    public int[] PivotArray(int[] nums, int pivot) 
    {
        
        /*
            use two pointers to traverse nums while also using two pointers to
            store nums[i] in a new array by simultaneously comparing pointers to pivot and
            updating pointers respectfully.
        */
        
        var PartedNumbers = new int[nums.Length]; 
        var left = 0;
        var right = nums.Length-1;
        
        int i,j;
                                                                        
        for (i = 0, j = nums.Length-1; i<nums.Length; i++, j--)   
        {
            var numberLeft = nums[i];
            var numberRight = nums[j];
            
            if (numberLeft < pivot)
            {
                PartedNumbers[left] = numberLeft;
                left++;
            } 
            
            if (numberRight > pivot)
            {
                PartedNumbers[right] = numberRight;
                right--;
            }
        }
        
        while (left <= right)
        {
            PartedNumbers[left] = pivot;
            
            left++;
        }
        
        return PartedNumbers;
    }
}

File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.rb
# @param {Integer[]} nums
# @param {Integer} pivot
# @return {Integer[]}
def pivot_array(nums, pivot)
    
    new_array = []
    
    a = 0
    until a === nums.length
        if nums[a] < pivot
            new_array.push(nums[a])
        end
        a+=1
    end
    
    b = 0
    until b >= nums.length
        if nums[b] === pivot
            new_array.push(nums[b])
        end
        b+=1
    end
    
    c = 0
    until c >= nums.length
        if nums[c] > pivot
            new_array.push(nums[c])
        end
        c+=1
    end
    
    new_array
end